### PR TITLE
4398 apply again after production delete old code

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

We have now switched on `apply_again_with_three_choices` in production. This is the cleanup.

## Changes proposed in this pull request

- Remove if else and unless statements (feature flag logic)
-  Remove application_submitted_apply_again email (superseded with `application_submitted` email)
- Remove duplicate tests (which have the feature flag off)
- Rework tests to suit the `apply_again_with_three_choices flow`
- Remove activation and deactivation of feature flag
- Refactor `candidate_helper.rb`

## Guidance to review

- Are there any more unused bits of code that we can remove?
- Are there any further opportunities to refactor?

## Link to Trello card

https://trello.com/c/dWqEYKjO/4398-apply-again-after-production-delete-old-code-and-remove-feature-flag

### CODE REVIEW

File: app/models/application_form.rb [hunk 1]
```diff
@@ -209,9 +209,7 @@ def candidate_has_previously_applied?
   end
 
   def candidate_can_choose_single_course?
-    return false if FeatureFlag.active?(:apply_again_with_three_choices)
-
-    apply_2?
+    false
```
Comment (frankieroberto) [inline app/models/application_form.rb:212]: Drop this whole method?

File: app/models/application_form.rb [hunk 2]
```diff
@@ -317,13 +315,7 @@ def support_cannot_add_course_choice?
   end
 
   def maximum_number_of_course_choices
-    if FeatureFlag.active?(:apply_again_with_three_choices)
-      MAXIMUM_NUMBER_OF_COURSE_CHOICES
-    elsif apply_1?
-      MAXIMUM_PHASE_ONE_COURSE_CHOICES
```
Comment (frankieroberto) [inline app/models/application_form.rb:323]: Delete the definition of these constants too?

File: app/models/application_form.rb (hunk 1 above)
Comment (avinhurry) [reply app/models/application_form.rb]: Thanks Frankie, i was originally going to refactor in a seperate PR but i've just done it here.

File: app/models/application_form.rb (hunk 2 above)
Comment (avinhurry) [reply app/models/application_form.rb:323]: Done.

**Review by stevehook — COMMENTED** [review]
This all looks good to me. I _think_ there may be a mail preview action that needs deleting (see inline comment) but otherwise everything seems to tie up.

File: app/services/candidate_interface/add_or_update_course_choice.rb [hunk 3]
```diff
@@ -58,16 +58,7 @@ def add
...
-        unless FeatureFlag.active?(:apply_again_with_three_choices)
-          course_choices = application_form.application_choices
-          flash[:success] = "You’ve added #{course_choices.last.course.name_and_code} to your application"
-        end
-
-        if application_form.can_add_more_choices? && !FeatureFlag.active?(:apply_again_with_three_choices)
-          redirect_to candidate_interface_course_choices_add_another_course_path
-        else
-          redirect_to candidate_interface_course_choices_review_path
-        end
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/services/candidate_interface/add_or_update_course_choice.rb:70]: The content on the `another` page wasn't helpful in the context of low apply again rates, and the design change we made in #6493 essentially made the page feel unnecessary.

File: app/mailers/candidate_mailer.rb [hunk 4]
```diff
@@ -11,15 +11,6 @@ def application_submitted(application_form)
...
 
-  def application_submitted_apply_again(application_form)
-    @application_choice = application_form.application_choices.first
-    @reject_by_default_date = @application_choice.reject_by_default_at.to_s(:govuk_date)
-
-    email_for_candidate(
-      application_form,
-    )
-  end
-
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/mailers/candidate_mailer.rb:22]: Good catch! 👀  Thank you. Have deleted ♻️

File: app/services/feature_flag.rb [hunk 5]
```diff
@@ -40,7 +40,6 @@ def feature
     [:publish_monthly_statistics, 'Publish monthly statistics at publications/monthly-statistics', 'Duncan Brown'],
     [:draft_vendor_api_specification, 'The specification for Draft Vendor API v1.1', 'Abeer Salameh'],
     [:immigration_entry_date, 'Extends the restructured_immigration_status feature to include the "Date of entry into the UK" question', 'Steve Hook'],
-    [:apply_again_with_three_choices, 'Replaces the current apply again logic which only allows one choice', 'Avin & Tomas'],
```
[the remark this answers is not in the document]
Comment (avinhurry) [reply app/services/feature_flag.rb:43]: Discussed offline: The decission is to revert this change and make it in a seperate PR where we will also delete the feature flag in the database.

**Review by tomas-stefano — APPROVED** [review]

File: app/controllers/candidate_interface/after_sign_in_controller.rb [hunk 6]
```diff
@@ -13,7 +13,7 @@ def interstitial
         flash[:warning] = "You have already selected #{course_from_find.name_and_code}."
         redirect_to candidate_interface_course_choices_review_path
       elsif current_application.maximum_number_of_course_choices?
-        error_message_key = current_application.apply_1? ? 'errors.messages.too_many_course_choices' : 'errors.messages.apply_again_course_already_chosen'
+        error_message_key = 'errors.messages.too_many_course_choices'
```
Comment (frankieroberto) [inline app/controllers/candidate_interface/after_sign_in_controller.rb:16]: Minor: maybe move this directly into the `I18n.t` method below and skip the variable now?

**Review by frankieroberto — APPROVED** [review]

File: app/controllers/candidate_interface/after_sign_in_controller.rb (hunk 6 above)
Comment (avinhurry) [reply app/controllers/candidate_interface/after_sign_in_controller.rb]: Good shout. Thanks - have updated ♻️

**Pull request merged**